packages/grid/_modules_/grid/components/menu/columnMenu/SortMenuItems.tsx

```diff
@@ -0,0 +1,56 @@
+import { MenuItem } from '@material-ui/core';
+import * as React from 'react';
+import { useGridSelector } from '../../../hooks/features/core/useGridSelector';
+import { sortModelSelector } from '../../../hooks/features/sorting/sortingSelector';
+import { ColDef } from '../../../models/colDef/colDef';
+import { SortDirection } from '../../../models/sortModel';
+import { ApiContext } from '../../api-context';
+
+export interface SortMenuItemsProps {
+  column: ColDef;
+  onClick: (event: React.MouseEvent<any>) => void;
+}
+
+export const SortMenuItems: React.FC<SortMenuItemsProps> = ({ column, onClick }) => {
+  const apiRef = React.useContext(ApiContext);
+  const sortModel = useGridSelector(apiRef, sortModelSelector);
+
+  const sortDirection = React.useMemo(() => {
+    if (!column) {
+      return null;
+    }
+    const sortItem = sortModel.find((item) => item.field === column.field);
+    return sortItem?.sort;
+  }, [column, sortModel]);
+
+  const onSortMenuItemClick = React.useCallback(
+    (event: React.MouseEvent<HTMLElement>) => {
+      onClick(event);
+      const direction = event.currentTarget.getAttribute('data-value') || null;
+      apiRef?.current.sortColumn(column!, direction as SortDirection);
+    },
+    [apiRef, column, onClick],
+  );
+
+  if (!column || !column.sortable) {
+    return null;
+  }
+
+  return (
+    <React.Fragment>
+      <MenuItem onClick={onSortMenuItemClick} disabled={sortDirection == null}>
+        Unsort
+      </MenuItem>
+      <MenuItem onClick={onSortMenuItemClick} data-value={'asc'} disabled={sortDirection === 'asc'}>
```

Comment:
This would be simpler and as performant

```suggestion
      <MenuItem onClick={onSortMenuItemClick('asc')} disabled={sortDirection === 'asc'}>
```